data/functions/calendar.sql

```diff
@@ -0,0 +1,220 @@
+                'Multi-state',
+                'Held Today'
+            ], ' ')
+        when array_length(contest, 1) = 0 then array_to_string(
+            array[
+                party,
+                office_sought,
+                expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
+                'Held Today'
+            ], ' ')
+        else array_to_string(
+            array[
+                array_to_string(contest, ', ') || ':',
+                party,
+                office_sought,
+                expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
+                'Held Today'
+            ], ' ')
+        end;
+    end
+$$ language plpgsql;
+
+
+-- Trying to make the names flow together as best as possible
+-- Like:
+    -- FL: House General Election Held Today
+    -- NH, DE: DEM Convention Held Today
+    -- General Election Held today States: NY, CA, FL, LA
+create or replace function generate_election_summary(trc_election_type_id text, office_sought text, contest text[], party text, trc_election_id numeric)
+returns text as $$
+    begin
+        return case
+        when array_length(contest, 1) > 3 then array_to_string(
+            array[
+                party,
+                office_sought,
+                expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
+                'Held Today',
+                'Contests:',
+                array_to_string(contest, ', ')
+            ], ' ')
+        when array_length(contest, 1) = 0 then array_to_string(
+            array[
+                party,
+                office_sought,
+                expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
+                'Held Today'
+            ], ' ')
+        else array_to_string(
+            array[
+                array_to_string(contest, ', ') || ':',
+                party,
+                office_sought,
+                expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
+                'Held Today'
+            ], ' ')
+        end;
+    end
+$$ language plpgsql;
+
+
+
+-- Not all report types are on dimreporttype, so for the reports to all have
+-- titles, I am adding a case. Ideally, we would want the right mapping.
+create or replace function generate_report_description(office_sought text, report_type text, rpt_tp_desc text, contest text[])
+returns text as $$
+    begin
+        return case
+            when rpt_tp_desc is null and array_length(contest, 1) > 3 then
+                array_to_string(
+                array[
+                    expand_office_description(office_sought),
+                    report_type,
+                    'Report Multi-state Due Today'
```

Comment:
Would it be more human readable if this said `Report (for Multiple States) is Due Today` or `Report is Due Today in Multiple States`